File: 1103-distribute-candies-to-people/1103-distribute-candies-to-people.py
class Solution:
    def distributeCandies(self, candies: int, num_people: int) -> List[int]:
      result = [0] * num_people
      count = 1

      while candies > 0:
        give = min(count, candies)

        result[(count - 1) % num_people] += give
        candies -= give
        count += 1
      
      return result

File: 1103-distribute-candies-to-people/1103-distribute-candies-to-people.js
/**
 * @param {number} candies
 * @param {number} num_people
 * @return {number[]}
 */
var distributeCandies = function(candies, num_people) {
  const result = new Array(num_people).fill(0);

  let count = 1;

  while (candies > 0) {
    const give = Math.min(count, candies);

    result[(count - 1) % num_people] += give;
    candies -= give;
    count++;
  }

  return result;
};
